src/core/include/iDynTree/Core/MatrixView.h

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright (C) 2020 Fondazione Istituto Italiano di Tecnologia
+ *
+ * Licensed under either the GNU Lesser General Public License v3.0 :
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ * or the GNU Lesser General Public License v2.1 :
+ * https://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * at your option.
+ */
+
+#ifndef IDYNTREE_MATRIX_VIEW_H
+#define IDYNTREE_MATRIX_VIEW_H
+
+#include <cassert>
+
+#include <type_traits>
+
+#include <iDynTree/Core/Span.h>
+#include <iDynTree/Core/Utils.h>
+
+// constexpr workaround for SWIG
+#ifdef SWIG
+#define IDYNTREE_CONSTEXPR
+#else
+#define IDYNTREE_CONSTEXPR constexpr
+#endif
+
+
+namespace iDynTree
+{
+
+    // this is required to be compatible with c++17
+    template <typename... Ts> struct make_void
+    {
+        typedef void type;
+    };
+    template <typename... Ts> using void_t = typename make_void<Ts...>::type;
```

Comment:
We may need this somewhere else in the future. Consider putting it in a namespace specific to MatrixView.

Reply:
```cpp
namespace MatrixViewInternal
{
    template <typename... Ts> struct make_void
    {
        typedef void type;
    };
    template <typename... Ts> using void_t = typename make_void<Ts...>::type;
}
```
is it fine?

Reply:
Yes!

Reply:
Namespace added here: https://github.com/robotology/idyntree/pull/734/commits/d766d55356134917ccbc1478286f8b250a02dc73